instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/RequestMetrics.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.context.Context;
+
+public interface RequestMetrics {
+
+  Context start(Context context, Attributes requestAttributes);
+
+  void end(Context context, Attributes responseAttributes);
```

Comment:
My understanding is metrics will only ever be recorded with "request attributes" since metric attributes (labels) must be constant throughout a request.

But some metrics will need response attributes to get the actual value, for example, a "response content length" metric. In other words, some span attributes (usually response attributes) are actually metric values.